src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyOperationManager.cs

```diff
@@ -403,6 +416,32 @@ public virtual TestProcessStartInfo UpdateTestProcessStartInfo(TestProcessStartI
         return updatedRunSettingsXml;
     }
 
+    private bool IsTesthostCompatibleWithTestSessions()
+    {
+        // These constants should be kept in line with the friendly names found in
+        // DotnetTestHostManager.cs, respectively DefaultTestHostManager.cs.
+        //
+        // We agreed on checking the test session compatibility this way (i.e. by reading the
+        // friendly name and making sure it's one of the testhosts we control) instead of a more
+        // generic alternative that was initially proposed (i.e. by decorating each testhost
+        // manager with a capability attribute that could tell us if the test session scenario
+        // is supported for the testhost in discussion) because of the breaking risks associated
+        // with the latter approach. Also, there is no formal specification for now of what it
+        // means to support test sessions. Should extending session functionality to 3rd party
+        // testhosts be something we want to address in the future, we should come up with such
+        // a specification first.
+        const string dotnetTesthostFriendlyName = "DotnetTestHost";
+        const string defaultTesthostFriendlyName = "DefaultTestHost";
```

Comment:
Could we make these fields/properties so we can add tests to ensure they are in sync?

Reply:
Done, thanks.